### ISSUE TITLE
Better UI : display axios errors nicely

### ISSUE THREAD

**Issue by benoit74** [openzim/kolibri#70]
With the new UI, axios errors are handled but not displayed.

We should :
- display the "nice" error to the user
- be able to open a debug UI where we could see all errors (navigator console?)

**Comment by Sudarshan-21** [comment]
@benoit74 can you provide me with some testases for the errors we are expecting, OR can you provide me with the steps to produce these errors in the code for testing purpose.

**Comment by benoit74** [comment]
Could be any kind of error, so we have to be ready for it. These errors are for now logged with `console.log` statements, but we need to display them nicely in the UI.
Most probably situation is a 404 error because of a bad request, typically a request to a topic which does not exists in the ZIM. Simple to produce by loading a topic which does not exists inside the ZIM.
But code should "cope" with all other errors as well, so it should probably quite generic. It is not expected to happen at all, but it could. I suggest to display something like "An error happened, cannot load requested content" with a "More details" button to display more details : HTTP return code and reply body if any ; or javascript error.

**Comment by Sudarshan-21** [comment]
I don't exactly seem to find any `console.log`s in the new_ui branch for any axios errors. but I encountered the axios errors taking care of errors concerned with `fetchchannel()` and `fetchtopic()` and I took care of those two errors (screenshot attached).
But as you said in the comment that the axios errors are getting logged into console, I do not seem to find those anywhere (have a look at the sidebar in the following screenshot)

![Screenshot from 2024-03-28 14-36-25](https://github.com/openzim/kolibri/assets/137636943/e2b52e50-ea1c-4440-afa1-c35c1877d5d4)

**Comment by benoit74** [comment]
You are right, we probably decided to stop logging these errors, anyway, you found "the place".

**Comment by aditya955569** [comment]
has this been fixed?

**Comment by benoit74** [comment]
Yes, indeed, fixed by #118
